app/models/description.rb

```diff
@@ -566,10 +522,61 @@ def update_users_and_parent
       SiteData.update_contribution(:del, editors_join_table, user.id)
     end
 
+    return unless parent.description_id == id
+
     # Make sure parent doesn't point to a nonexisting object.
-    if parent.description_id == id
-      parent.description_id = nil
-      parent.save_without_our_callbacks
+    parent.description_id = nil
+    parent.save_without_our_callbacks
+  end
+
+  ##############################################################################
+
+  private
+
+  # Descriptive subtitle for this description (when it is not necessary to
+  # include the title of the parent object), in plain text.  [I'm not sure
+  # I like this here.  It might violate MVC a bit too flagrantly... -JPH]
+  def put_together_name(full_or_part)
+    tag = :"description_#{full_or_part}_title_#{source_type}"
+    user_name = begin
+                  user.legal_name
+                rescue StandardError
+                  "?"
+                end
+    args = {
+      text: source_name,
+      user: user_name
+    }
+    if full_or_part == :full
+      args[:object] = parent.format_name
+    elsif source_name.present?
+      tag = :"#{tag}_with_text"
     end
+    tag.l(args)
+  end
+
+  # Change a given User's or UserGroup's privileges.
+  def chg_permission(groups, arg, mode)
+    arg = UserGroup.one_user(arg) if arg.is_a?(User)
+    if (mode == :add) &&
+       !groups.include?(arg)
+      groups.push(arg)
+    elsif (mode == :remove) &&
+          groups.include?(arg)
+      groups.delete(arg)
+    end
+  end
+
+  def user_made_a_change?(user)
```

Comment:
**Questions**: 
* Would `EXISTS` make the SQL more efficient? (This method just needs to return a truthy value; we don't need to load an `id`.) 
* If so, what's the syntax? (I couldn't figure it out.)